### PR TITLE
DOC/CI: include api docs on travis

### LINKED ISSUE #3800: add travis doc build
not sure if this is worth the trouble but it may catch version incompatibility issues and other bugs not caught by the test suite.

### LINKED ISSUE #6100: DOC/BLD: attributes initialized as None not picked up by autosummary anymore
I already mentioned it here: https://github.com/pydata/pandas/issues/5899#issuecomment-32967854, but then this did not appear on the dev doc builds on the website and only locally on my computer. But now I also see it there. See eg http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame in the list of attributes, the `is_copy` is no link anymore.

So I get the following warning: `DataFrame.is_copy.rst:: WARNING: document isn't included in any toctree` for 8 methods:
```
pandas.DataFrame.is_copy
pandas.DatetimeIndex.name
pandas.DatetimeIndex.offset
pandas.Index.asi8
pandas.Index.name
pandas.Panel.is_copy
pandas.Panel4D.is_copy
pandas.Series.is_copy
```

Strange thing is, there have not been recent changes to these functions (I think), but the errors only appeared recently. 
@y-p You did the most recent doc build on the website? (does this not happen automatically anymore?)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Continuation of #11069 (It seems I cannot reopen that PR because I already force pushed the branch ..)

Closes #3800

TODO:
- [x] remove printing of doc build log file in after_script

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Reason for reopening this, is that I noticed that sphinx 1.3 seemed to be improved in performance. And indeed, now the doc build on travis takes less than 20 minutes, while before it hit the maximum travis run time of 50 min.

**Comment by jorisvandenbossche** [comment]
@jreback Can you have a look at this? Note that there is still one temp commit that should be removed, so the full diff is not fully correct

This is related to https://github.com/pydata/pandas/issues/12003 (@shoyer), but I think this is still relevant. Maybe not necessarily for building the dev docs (if we start using RTD for this), but you can always check the doc build log for PRs this way.

File: ci/before_install.sh [hunk 1]
```diff
@@ -11,3 +11,7 @@ echo "inside $0"
 sudo apt-get update $APT_ARGS # run apt-get update for all versions
 
 true # never fail because bad things happened here
+
+if [ x"$DOC_BUILD" == x"" ]; then
+    exit 1
```
Comment (jreback) [inline ci/before_install.sh]: shouldn't this be exit 0 (otherwise its an error and will fail travis)

Comment (jreback) [inline ci/before_install.sh]: though maybe this was for testing?

**Comment by jreback** [comment]
```
./make.py
sh: 1: jupyter: not found
Running Sphinx v1.3.3
```

do you need to add jupyter to the install?

File: ci/before_install.sh (hunk 1 above)
Comment (jorisvandenbossche) [inline ci/before_install.sh]: yes, this was for testing

**Comment by jorisvandenbossche** [comment]
Rebased, removed the temporary commit, and added `nbconvert` as a dependency

**Comment by TomAugspurger** [comment]
FYI, I saw [nbsphinx](http://nbsphinx.readthedocs.org/en/0.2.3/) the other day. I ran it on our style notebook and it seemed to work pretty well. There were a few CSS things that don't get picked up properly, but we can probably move to that soon.

**Comment by jreback** [comment]
@jorisvandenbossche this looks good.

only thing (and can do later), is maybe move this build up so its the first one after main builds (before slow ones).

**Comment by jreback** [comment]
merge when ready (try out ``merge-pr.py``)

**Comment by jreback** [comment]
@jorisvandenbossche 

build on master: https://travis-ci.org/pydata/pandas/jobs/104512895

failed apparently

**Comment by jorisvandenbossche** [comment]
Thanks for noticing. Probably due to the adding of nbconvert to requirements before I merged (what will maybe have given an incompatible nbconvert and ipython version, will need to investigate that further). 
I actually already failed in the PR, but because it's one of the allowed failures, I didn't notice ...

https://github.com/pydata/pandas/pull/12138

**Pull request closed without merging**